### PR TITLE
Return set of positive actions after each step (#7)

### LINKED ISSUE #7: Return set of "positive" actions in the step() function's info return value
# Why?

The base paper states that using a "guided exploration strategy following the principles of apprenticeship learning" does help to speed up the exploration phase during the agent's training phase.

This exploration strategy relies on "Identify[ing] which actions will give positive and negative rewards", which is possible since "the environment knows the ground truth boxes, and the reward function is calculated with respect to the IoU with the current box". 

Thus, the agent has to be "allowed to choose one random action from the set of positive actions[1]". 
[1] "Or any action if this set of positive actions is empty."

# What?

In order to allow the agent to randomly select one action **from the set of positive actions** during training, the environment should return this set (of positive functions, i. e. actions that give a positive reward) as part of the `step()` functions's `info` return value:

```python
    def step(self, action):
        """Execute an action and return
            state - the next state,
            reward - the reward,
            done - whether a terminal state was reached,
            info - any additional info"""
```

(Notice the last line of the doc comment)

### PR DESCRIPTION
**Pull request by J-L-O**
Closes #7

### CODE REVIEW

File: text_localization_environment/TextLocEnv.py [hunk 1]
```diff
@@ -70,12 +83,31 @@ def step(self, action):
...
+
+        return reward
+
+    def find_positive_actions(self):
+        rewards = np.array([self.calculate_potential_reward(i) for i in self.action_set])
+
+        positive_actions = np.arange(0, self.action_space.n)[rewards > 0]
+
+        if len(positive_actions) == 0:
+            return np.arange(0, self.action_space.n)
```
Comment (bakoe) [inline text_localization_environment/TextLocEnv.py]: Here you return a `ndarray`, whereas in line 110 you return a `list`. You might choose one type for both of the return statements.

**Review by bakoe — APPROVED** [review]
Very cool, thanks for doing this! LGTM and I can somehow confirm that it's working:

I've manually initialized an environment and chose randomly one of the ”positive” actions returned in the `info` parameter each time. The result is as follows:

info_return_video

**Comment by bakoe** [comment]
image

I cases like this, should we maybe return the set of actions that “center the bounding box” over the target bounding box better? I know that based on the strict IoU calculation formula, in the scenario depicted in the screenshot no action improves the IoU – however, if we would target the agent towards centering its viewport (bbox) above the target bounding box, the result would probably improve. 

What do you think?

**Comment by J-L-O** [comment]
I think to some extend this is going to be resolved by #12 as that would result in a more nuanced reward structure.

File: text_localization_environment/TextLocEnv.py (hunk 1 above)
Comment (J-L-O) [reply text_localization_environment/TextLocEnv.py]: True, fixed

**Pull request merged**